### ISSUE TITLE
Possibly to choose between degree and  "gardian" as angle unit

### ISSUE THREAD

**Issue by MartinMap** [qgis/QGIS#47447]
### Feature description

It would be great to have the possibly to choose between degree and [gardians](https://en.wikipedia.org/wiki/Gradian) as angle unit. CAD users in particular would be happy about this!

**Comment by uclaros** [comment]
You can select grads in qgis by going to: `Setings > Options > Map Tools > Preferred angle units` and picking `gon/gradians`, but it only applies to the measure tools.
It would probably be useful to also change the units for the advanced digitizing dock tools.
@MartinMap Are there other places where changing the angle unit is needed?

**Comment by MartinMap** [comment]
Yes, especially while digitizing with the Advanced Digitizing Panel or rotating something. Gon for measurement is nice, but it could also be very important for construction. (a lot of people are switching  from CAD to QGIS and would enjoy that)
![Selection_044](https://user-images.githubusercontent.com/6401482/154964757-0cfcd49e-ca5c-4919-a2d5-4d9871d5888b.png)

**Comment by uclaros** [comment]
Quick tip for the advanced digitizing tools until this is properly implemented: 
All advanced digitizing fields support arithmetic operators, so you can type `{angleInGrads} * 90 / 100` to get your angle in degrees! So if for example you have a *42 grad* angle you can type in the angle field `42*.9` and press enter to get *37.8* which is your angle in degrees.
